core/src/main/java/com/redhat/lightblue/client/response/LightblueResponse.java

```diff
@@ -101,6 +101,11 @@ private int parseInt(String fieldName) {
         }
 
         try {
+            JsonNode errorCode = json.path("errorCode");
+            if (errorCode != null && !errorCode.isNull() && !errorCode.isMissingNode()){
+                throw new LightblueResponseParseException("Error parsing lightblue response: Error returned. JSON returned:" + json.toString() + "\n");
+            }
+
```

Comment:
I will try to make a fix o the hasError method, which would include to check errorCode in the json response and I also thought it would be nice to check the HTTP status codes (mainly because if we got the deployment not running and the application server or even the load balancer could return an error and that would not be noticed but that method right now )